### ISSUE TITLE
fix(eval): three NLP golden tests fail because their label files are unpublishable

### ISSUE THREAD

**Issue by VforVitorio** [VforVitorio/F1_Strat_Manager#1130]
## Symptom

Three tests in `tests/eval/test_nlp_golden.py` fail on this machine and cannot pass on any machine:

```
FAILED test_intent_reproduction_is_sane        - assert 'pending' == 'reproduced'
FAILED test_alert_precision_from_gold_is_high  - assert 'pending' == 'reproduced'
FAILED test_ner_entity_f1_reproduces_headline  - assert 'pending' == 'reproduced'
```

They are the last three of the nine local failures on `dev` now that the full `data/raw` tree is on disk.

## Cause

Each of the three is gated by a `skipif` that probes the **model weights** and never the **labels**:

| test | skipif probes | the stage actually also reads |
|---|---|---|
| `test_intent_reproduction_is_sane` | `intent_setfit_modernbert_v1/model_head.pkl` | `data/processed/radio_nlp/intent_labeled_data.csv` (`nlp.py:221`) |
| `test_alert_precision_from_gold_is_high` | `intent_setfit_modernbert_v1/model_head.pkl` | the same CSV |
| `test_ner_entity_f1_reproduces_headline` | `ner_v1/bert_bio_v1/bert_bio_state_dict.pt` | `data/processed/radio_nlp/f1_radio_entity_annotations.json` (`nlp.py:518`) |

The three stages return a `pending` row when the labels are absent, and the tests assert `reproduced`. So they skip on CI, where there is no `data/` at all and the model probe is already False, and they fail on exactly the machine that has the weights but not the labels. That is every install, because neither label file is downloadable.

Verified against the Hub API on `VforVitorio/f1-strategy-dataset`: `find --path "*radio_nlp*"`, `find --name "*annotations*"` and `find --name "*intent_labeled*"` all return **zero entries**. Locally `data/processed/radio_nlp/` holds only `2025/{australia,monaco,qatar}/transcripts.json`.

## The two artefacts are not in the same situation

**`intent_labeled_data.csv` exists and is already in git**, at `legacy/radio_label/2023/intent_labeled_data.csv`: 56.4 KB, columns `message,intent`, **529 rows**, which is the exact count `_intent_predictions`' own docstring quotes. Copying it to `data/processed/radio_nlp/` makes both intent tests pass in 11.7 s (executed). So it is one publish away from real, and the two tests that need it are recoverable rather than dead.

**`f1_radio_entity_annotations.json` exists nowhere.** Not on the Hub, and a repo-wide search for `*annotation*` outside `.venv` returns nothing. `test_ner_entity_f1_reproduces_headline` has no path to green until that file is produced or found.

## Fix

Two parts, and only the first is in this PR's scope.

1. **Now: gate the three on the labels too**, so they skip for the reason that is true instead of failing. Probe through `get_data_root()` rather than `ROOT / "data"`, because `F1_STRAT_DATA_ROOT` moves the tree and a probe that does not read what the code reads is the class of bug this repo keeps paying for.
2. **Then: publish `intent_labeled_data.csv`** to the dataset repo and add it to `_DEFAULT_PROCESSED_PATTERNS`, which turns two of the three skips back into real measurements. Same shape as #798 and #837: an artefact the code needs that the download never asks for. The NER annotations need a separate decision, because there is nothing to publish.

## Guard

The skip itself is the guard, and its reason string names the missing file. A test that asserts the artefact is absent would be asserting the bug.

Note for whoever picks up part 2: the other five probes in the file also read `ROOT / "data"`. With `F1_STRAT_DATA_ROOT` set they degrade to a false **skip**, not a false failure, which is why they are left alone here.
